contracts/utils/Funder.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+
+// I wanted to build this in a way that you can also use it to fund 100 sponsor
+// wallets that each power a single self-funded Beacon.
+// Use FunderFactory to deploy.
+contract Funder is Ownable {
+    // Making this immutable would be nice. However, I foresee that we will
+    // want to update this and having to transfer funds to do so will be very
+    // annoying/error-prone.
+    bytes32 public root;
+
+    // This contract is intended to be cloned so the constructor makes the
+    // implementation unusable
+    constructor() {
+        _transferOwnership(address(0));
+        root = bytes32(uint256(1));
```

Comment:
Was this something intended to initialize `root` with `1`? Probably I'm missing sth.

Reply:
I was wondering the same thing, it feels like `root` should be initialized with `bytes32(0)` or `0x00`

Reply:
It's so that the implementation can't be initialized

Reply:
To make it clearer, we're using cloning method to deploy the Funder.sol contract. The constructor does not execute during deployment, and the contract is initialized directly. So `root == bytes32(0)` will be validated when initializing. This makes the deployment more efficient and saves gas.

Reply:
why not something like:
```
root = bytes32(bytes("CANNOT_INITIALIZE"));
```

Reply:
Would work just as well